src/picongpu/CMakeLists.txt

```diff
@@ -254,6 +254,9 @@ if( (Boost_VERSION EQUAL 105500) AND
       "${CUDA_NVCC_FLAGS} \"-DBOOST_NOINLINE=__attribute__((noinline))\" ")
 endif()
 
+# PMacc (ab)uses boost::result_of for non-functors. This forces boost to use the
+# result<> member template of the target type if available
+add_definitions(-DBOOST_RESULT_OF_USE_TR1)
```

Comment:
weird, looks like [1.56.0+](https://github.com/boostorg/utility/commit/2637dfcc592c0a1e985ac889e1e589fd469801d2#diff-d19580c8f35d7254546177794b54ccb9) from the git history... [at least 1.52.0+](http://lists.boost.org/Archives/boost/2012/09/196039.php)